internal/kldevents/eventstream.go

```diff
@@ -304,16 +398,30 @@ func (a *eventStream) batchDispatcher() {
 					return
 				}
 				currentBatch = append(currentBatch, event)
+			case <-a.updateInterrupt:
+				// we were notified by the caller about an ongoing update, cancel the timeout ctx and return
+				log.Infof("%s: Notified of an ongoing stream update, will not dispatch batch", a.spec.ID)
+				a.updateWG.Done()
```

Comment:
Should this just be a `defer`? Does this fulfil the same purpose as `dispatcherDone`

Reply:
moved this and other calls to a `defer`. Left invocation in `batchProcessor` as is since we need to unlock after calling `Done()` and that's not in all cases before returning